### PR TITLE
CRB 259/minimum difficulty

### PR DESCRIPTION
**Pull request by CAGS295**
depends on #12

### CODE REVIEW

File: src/engine.rs [hunk 1]
```diff
@@ -84,3 +66,38 @@ impl Engine for PowEngine {
...
+    let mut node: PowNode = match self.config.take() {
+      Some(config) => PowNode::with_config(config, service),
+      None => PowNode::new(service),
+    };
+
+    // Initialize the PoW based on the current startup state received from the
+    // validator - an error here is considered fatal and prevents startup.
+    //
+    // Note: Errors from this call don't propagate due to conflicting types,
+    // this means we need to handle them explicity.
```
Comment (nathanwhit) [inline src/engine.rs]: I'm confused by this comment, the types of the error _do_ match up. Replacing this statement with just 
```rust
node.initialize(startup)?;
Ok(node)
```
compiles and runs correctly.

File: src/miner/miner.rs [hunk 2]
```diff
@@ -42,7 +47,7 @@ impl Miner {
     config: &PowConfig,
   ) -> Result<(), Error> {
     let block: Block = service.get_block(&block_id)?;
-    let header: BlockHeader = BlockHeader::borrowed(&block).expect("Block Header");
+    let header: BlockHeader = BlockHeader::borrowed(&block).expect("Block head Header");
```
Comment (nathanwhit) [inline src/miner/miner.rs]: I'm a little confused by "block head header" is it a typo, or what is a "block head"?

**Review by nathanwhit — COMMENTED** [review]
A couple comments but overall looks good to me. Should be good to go once #10 is merged.

File: src/engine.rs (hunk 1 above)
Comment (CAGS295) [reply src/engine.rs]: Good catch, these are old comments.

File: src/miner/miner.rs (hunk 2 above)
Comment (CAGS295) [reply src/miner/miner.rs]: Yup, that's not very descriptive. What about Chain head header?

**Review by nathanwhit — DISMISSED** [review]
Looks good to me

**Review by atodorov — CHANGES_REQUESTED** [review]
Needs rebase b/c there are some conflicts.

**Review by atodorov — APPROVED** [review]
After discussion at https://gluwa.slack.com/archives/C7JJP9WTF/p1637333833199600 we agreed to merge this PR.

**Review by nathanwhit — APPROVED** [review]

**Pull request merged**